aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1877,6 +1753,146 @@ impl VMAdapter for AptosVM {
     }
 }
 
+// Executor external API
+impl VMExecutor for AptosVM {
+    /// Execute a block of `transactions`. The output vector will have the exact same length as the
+    /// input vector. The discarded transactions will be marked as `TransactionStatus::Discard` and
+    /// have an empty `WriteSet`. Also `state_view` is immutable, and does not have interior
+    /// mutability. Writes to be applied to the data view are encoded in the write set part of a
+    /// transaction output.
+    fn execute_block(
+        transactions: &[SignatureVerifiedTransaction],
+        state_view: &(impl StateView + Sync),
+        maybe_block_gas_limit: Option<u64>,
+    ) -> Result<Vec<TransactionOutput>, VMStatus> {
+        fail_point!("move_adapter::execute_block", |_| {
+            Err(VMStatus::error(
+                StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
+                None,
+            ))
+        });
+        let log_context = AdapterLogSchema::new(state_view.id(), 0);
+        info!(
+            log_context,
+            "Executing block, transaction count: {}",
+            transactions.len()
+        );
+
+        let count = transactions.len();
+        let ret = BlockAptosVM::execute_block::<
+            _,
+            NoOpTransactionCommitHook<AptosTransactionOutput, VMStatus>,
+        >(
+            Arc::clone(&RAYON_EXEC_POOL),
```

Comment:
Again, maybe part of a different refactoring, but we had this super long TODO not to expose RAYON_EXEC_POOL in aptos_vm.rs, but have it exist in BlockAptosVM (less chance for using it unexpectedly @sitalkedia @ziaptos). Maybe it's a good time to do that, or we can add a TODO.